feathr_project/feathr/client.py

```diff
@@ -946,19 +947,32 @@ def _collect_secrets(self, additional_secrets=[]):
             prop_and_value[prop] = self.envutils.get_environment_variable_with_default(prop)
         return prop_and_value
 
-    def get_features_from_registry(self, project_name: str) -> Dict[str, FeatureBase]:
+    def get_features_from_registry(self, project_name: str, return_keys: bool = False, verbose: bool = False) -> Union[Dict[str, FeatureBase], tuple]:
         """
         Get feature from registry by project name. The features got from registry are automatically built.
         """
         registry_anchor_list, registry_derived_feature_list = self.registry.get_features_from_registry(project_name)
         self.build_features(registry_anchor_list, registry_derived_feature_list)
         feature_dict = {}
+        key_dict = {}
         # add those features into a dict for easier lookup
+        if verbose and registry_anchor_list:
+            logger.info("Get anchor features from registry: ")
         for anchor in registry_anchor_list:
             for feature in anchor.features:
                 feature_dict[feature.name] = feature
+                key_dict[feature.name] = feature.key
```

Comment:
we can assign key_dict only when `return_keys==True` since we don't use the var when it's false. You may use dict-comprehension too like:
```
feature_dict = {
    feature.name: feature for anchor in registry_anchor_list for feature in anchor.features
}
if return_keys:
    key_dict = {
        feature.name: feature.key for anchor in registry_anchor_list for feature in anchor.features
    }
```

Same to the derived feature list.

Reply:
> we can assign key_dict only when `return_keys==True` since we don't use the var when it's false.
> 
> Also if we treat the derived key the same, we can merge them together and do loop.
> 
> You may use dict-comprehension like:
> 
> ```
> anchor_list = registry_anchor_list + registry_derived_feature_list
> 
> feature_dict = {
>     feature.name: feature for anchor in anchor_list for feature in anchor.features
> }
> if return_keys:
>     key_dict = {
>         feature.name: feature.key for anchor in anchor_list for feature in anchor.features
>     }
> ```

this is a good point

Reply:
It can cause more time consumption since we need to go through both anchor list and derived features list again.